FAQ cleanup

foss- opened the pull request:
- updated client section
- updated contact states section

Commit:
updated contac states

- added sub-headlines for better readibility
- rephrased to match new format

Commit:
FAQ: improved contact state title

Commit:
FAQ: Client update

- remove Hotot (unmaintained / EOL)
- Choqok: added link
- AndStatus: added links
- Mustard-Mod (removed, looks dead https://github.com/royalterra/Mustard-Android-APK)
- Mustard (removed, also dead http://mustard.macno.org/
- Twidere added links
- DiCa (removed, dead https://dica.mixi.cool/)
- Fedilab added links
- Friendiqa added links

Commit:
FAQ: Client additional links

Commit:
renamed client section

"List of friendica clients" not a question, but more compact and easier to process

MrPetovan commented on doc/FAQ.md:
```diff
@@ -9,11 +9,11 @@ User
 * **[Can I subscribe to a hashtag?](help/FAQ#hashtag)**
 * **[How to create a RSS feed of the stream?](help/FAQ#rss)**
-* **[Are there any clients for friendica I can use?](help/FAQ#clients)**
+* **[List of friendica clients](help/FAQ#clients)**
```
Why did you change this from a question to a headline even though it's the style we've been using for the FAQ?

foss- replied:
Thought it to be more compact and quicker to grasp by users. If you want to stick with the question format, we can revert the client list title.

MrPetovan commented on doc/FAQ.md:
```diff
@@ -94,26 +94,25 @@ in German:
-**Ignored contacts** are included in delivery - they will receive your posts and private messages.
-However we do not import their posts or private messages to you.
-Like blocking, you will still see this person's comments to posts made by your friends.
+Direct communication will be blocked. Blocked contacts are not included in delivery, and their own posts to you are not imported. However their conversations with your friends will still be visible in your stream. If you remove a contact completely, they can send you another friend request. Blocked contacts cannot do this. They cannot communicate with you directly, only through friends.
```
Please add a new line after each sentence. It doesn't affect the display but makes this file easier to read and change. This goes for all Markdown files.

MrPetovan commented on doc/FAQ.md:
```diff
@@ -94,26 +94,25 @@ in German:
-However a hidden contact will appear normally in conversations and this may expose his/her hidden status to anybody who can see the conversation.
+An addon called "blockem" can be installed to collapse/hide all posts from a particular person in your stream if you desire complete blocking of an individual, including their conversations with your other friends.
+
+##### Archieved
```
Typo
```suggestion
##### Archived
```

MrPetovan commented on doc/FAQ.md:
```diff
@@ -94,26 +94,25 @@ in German:
-An **archived contact** means that communication is not possible and will not be attempted.
-(Perhaps the person moved to a new site and removed the old profile.)
-However unlike blocking, existing posts this person made before being archived will be visible in your stream.
+Not included in delivery but they will receive your posts and private messages. However we do not import their posts or private messages to you. Like blocking, you will still see this person's comments to posts made by your friends.
```
"Not included in delivery but they will receive your posts and private messages."

What delivery are they not included in?

Commit:
Update doc/FAQ.md

Co-authored-by: Hypolite Petovan <hypolite@mrpetovan.com>

Commit:
Update FAQ.md

- revert to question format for client list
- new line for each sentence as requested
- corrected Ignored contacts section: they are included in delivery and will receive posts and private messages

Commit:
Update FAQ.md

- Archived (typo)
